### ISSUE TITLE
Make type checking mypy more straightforward for new contributors

### ISSUE THREAD

**Issue by pranavrajpal** [python/mypy#11403]
There are a couple pitfalls relating to type checking mypy itself that don't seem to be documented anywhere:
1. Specifying files: Running `mypy .` and `mypy mypy mypyc` both result in duplicate module errors that prevent further type checking. The correct way to run mypy is `mypy -p mypy -p mypyc`, which doesn't appear to be documented anywhere (I only found this by looking at the tox.ini config file).
2. Specifying the config file: Failing to specify mypy_self_check.ini as the config file results in an error in mypy/bogus_type.py and means that people aren't using mypy's stricter type checking options.

I think the solution to 2 would be to rename mypy_self_check.ini to mypy.ini, because self checking mypy is significantly more common than compiling mypy with mypyc, so making mypy_self_check.ini the default config file seems reasonable.

For 1, we should at the very least document the command to run mypy in CONTRIBUTING.md. I think the ideal solution would involve changing the configuration file so that either `mypy .` or `mypy mypy mypyc` just works correctly, the latter of which seems to be possible from my experiments by turning on `namespace_packages` and excluding mypyc/test-data.

**Comment by hauntsaninja** [comment]
Thanks, I agree this isn't intuitive. Note that knowing to specify the config file is the bigger issue, since `mypy mypy mypyc --config-file mypy_self_check.ini ` works just fine.

I think renaming the file to mypy.ini might cause issues with tests or something. But at the very least, this should be more clearly documented, which I've attempted to do in #11413

**Comment by pranavrajpal** [comment]
> `mypy mypy mypyc --config-file mypy_self_check.ini` works just fine.

Yeah, I didn't realize this when I created the issue. I wasn't using the latest master so I didn't see the effects of #11258.

> I think renaming the file to mypy.ini might cause issues with tests or something.

I'm guessing you're referring to these test failures that occur when I rename mypy_self_check.ini to mypy.ini:
```
FAILED mypy/test/testargs.py::ArgSuite::test_coherence - AssertionError: {'build_type': 0, 'python_version': (3, 8), 'python_executable': '/usr/bin/python3', 'platform': 'linux', 'custo...
FAILED mypy/test/testargs.py::ArgSuite::test_executable_inference - AssertionError: assert (3, 6) == (3, 8)
FAILED mypy/test/testsamples.py::SamplesSuite::test_samples
FAILED mypy/test/testsamples.py::SamplesSuite::test_stdlibsamples
```

All of these seem fixable if we change the current working directory to some other folder when running these tests.
